src/datasets/load.py

```diff
@@ -689,19 +790,56 @@ class PackagedDatasetModuleFactory(_DatasetModuleFactory):
     def __init__(
         self,
         name: str,
+        config_name: Optional[str] = None,
         data_dir: Optional[str] = None,
         data_files: Optional[Union[str, List, Dict]] = None,
         download_config: Optional[DownloadConfig] = None,
         download_mode: Optional[Union[DownloadMode, str]] = None,
     ):
+        if config_name and not (data_dir or data_files):
+            raise ValueError(
+                "You should specify `data_dir` or `data_files` (with README.md inside) "
+                "when providing a `config_name`, otherwise resolving is not possible. "
+            )
+
         self.name = name
+        self.config_name = config_name if config_name else "default"
         self.data_files = data_files
         self.data_dir = data_dir
         self.download_config = download_config
         self.download_mode = download_mode
         increase_load_count(name, resource_type="dataset")
 
     def get_module(self) -> DatasetModule:
+        def find_readme_in_files(files):
```

Comment:
there should be only one single readme no ?

Reply:
I don't think we should allow the readme to be passed in data_files at all, or it can be confusing

Users should pass a dataset repository path containing a readme

Reply:
I wanted to do so at first actually but then it won't be aligned with loading from path.
I mean usually `load_dataset(local_dir_name)` and `load_dataset("imagefolder", data_dir="local_dir_name"` are the same. But this would break it.
But I don't really mind tbh

Reply:
I think load_dataset(local_dir_name) and load_dataset("imagefolder", data_dir="local_dir_name") have a subtle difference: one loads a dataset repository (including metadata and all) and the other uses a packaged builder with a specific data_dir. This difference makes sense to me, so I think we can remove find_readme_in_files